Sprint 1 result

IReallyHateLogins opened the pull request:
### I propose we move our "stable" version from sprint 1 results.
We should label it somehow. I think squash should be applied to merge. 
**We really need to work on our squash commit comment for this version**, write our features, comments, what's not supported and what to improve. And **carefully resolve conflicts.**
I think @Valbod-I-am-a-Swiftie should resolve conflicts, but it's up for discussion.
I strongly suggest that **every collaborator** review, comment and carefully approve this request. 

Again, we should merge with **detailed and correct comment.** And correct and detailed reviews are welcome. 

_I will review later today. There's no rush, we can discuss, edit and format this merge for at least a week._

dahah-babah committed:
Create prototype1

dahah-babah committed:
Delete prototype1

dahah-babah committed:
Add files via upload

dahah-babah committed:
Create prototype1

dahah-babah committed:
Merge branch 'develop' of https://github.com/Valbod-I-am-a-Swiftie/Post-client into develop

dahah-babah committed:
Create prototype1

dahah-babah committed:
Merge branch 'develop' of https://github.com/Valbod-I-am-a-Swiftie/Post-client into develop

Commit:
Added readme for testing purposes

Commit:
Expanded readme to test Github slack notifications on dev

dahah-babah committed:
update index.html

dahah-babah committed:
create nav menu

dahah-babah committed:
Merge branch 'develop' of https://github.com/Valbod-I-am-a-Swiftie/Post-client into develop

ValentinBodrov committed:
Added the sequence diagram

ValentinBodrov committed:
Merge branch 'develop' of https://github.com/Valbod-I-am-a-Swiftie/Post-client into develop

dahah-babah committed:
update main.css

snowzhop committed:
Spring-server BEGIN!

Commit:
Added gradle script to pull resources

ValentinBodrov committed:
Added new folder  with diagrams and reports

ValentinBodrov committed:
Deleted the unnecessary old folders with umls and interface

ValentinBodrov committed:
Added new UMLs for the 1st sprint in the new folder. Removed some unnecessary *.drawio files, combined them into one *.drawio file

dahahbabah committed:
Test vs code ^_^

dahahbabah committed:
Added auth branch
Create html/css files for auth page

snowzhop committed:
Https branch (#2)

* Common controller changed to RestController

* Created application.properties

* Created keystore

* Returned to Tomcat v9.0.24

* Changed baseName and version

* Added port number for HTTP

* Added new class for redirection from HTTP to HTTPS

IReallyHateLogins committed:
Initital report final ver. + documentation restructure

Added approved final version of initial report
Removed unnessesary imges, you can create them locallyfrom draw.io
Created some sort of structure in documentation folder
 * Folder for each sprint/period
 * inside
    - report file containing final report
    - uml.drawio with all diagrams
    - Additional images and resourses with appropriate names

ValentinBodrov committed:
User model branch (#3)

- created new user model class **./src/main/java/hello/UserModel.java** which contains key information about user
- added two new dependencies in **build.gradle**:
  - the first one serves for Java Persistence API (JPA) implementation
  - the second one serves for the supporting the virtual database
- added new methods in **HelloController.java** for GET- and POST-requests

IReallyHateLogins committed:
Feature undertow (#4)

* Changed from tomcat to undertow

Moving to undertow due to tomcat producing an annoying exeption in log on every request, and all keystorage problems on tomcat 9.0.25+

* Added server address to config

Added adddres to config to replace explicit "0.0.0.0" on http redirect configuration

snowzhop committed:
Added user password to model

IReallyHateLogins committed:
Merge pull request #5 from Valbod-I-am-a-Swiftie/fix-user-model

Added user password to model

Egridnev committed:
IMAPConnector and SMTPConnector added (#6)

* IMAPConnector and SMTPConnector added

* Update build.gradle

remove unnecessary dependence

dahah-babah committed:
Created auth form.
Added authform styles.

IReallyHateLogins committed:
Merge pull request #8 from Valbod-I-am-a-Swiftie/auth

Auth

IReallyHateLogins committed:
Fixed formatting

IReallyHateLogins committed:
Merge pull request #9 from Valbod-I-am-a-Swiftie/mapping_branch

Mapping branch

ValentinBodrov committed:
Added GET-mapping for unique user with ID

IReallyHateLogins committed:
Basic mail functions (#10)

* Started to send mail

* Added class PostMessage to store message. Rewrite function getMailList in IMAPConnector

* Added function to send email

Function that sends email and saves it thru imap. Also changed mail wrapper to use strings instead of internal javax classes.

* Added send and delete function

Also added some endpoints for testing

* Fixed inaccuracies in resolving conflicts

* Upgrade PostMessage class.

Added fields numberInFolder, recipients folder.
Added correctly return of multipart content.

IReallyHateLogins committed:
Format fix branch (#11)

* Fixed formatting

* Updated gradle build script

ValentinBodrov committed:
Update PUT- and DELETE-mappings, added responce statuses and exception handling for every query

Commit:
Fixed gradle buildscript

Using resource path, it was impossible to set different dropoff for resources. Instead, I remade copy task, now it copies directly to build folder, with correct path.

ValentinBodrov committed:
Merge pull request #13 from Valbod-I-am-a-Swiftie/fix-buildscript

Fixed gradle buildscript

dahah-babah committed:
Removed all documentation

dahah-babah committed:
Merge pull request #14 from Valbod-I-am-a-Swiftie/fix-remove-documentation

Removed all documentation

Valbod-I-am-a-Swiftie commented on README.md:
```diff
@@ -0,0 +1,17 @@
+#### Current features
+
+- None yet
+- But we have some uml-s and web-gui design ideas!
```
I think we have some features for now, so it could be complited

Valbod-I-am-a-Swiftie commented on Spring-server/build.gradle:
```diff
@@ -0,0 +1,52 @@
+    compile "org.springframework.boot:spring-boot-starter-web"
+    compile 'org.springframework.boot:spring-boot-starter-undertow'
+    compile 'javax.mail:javax.mail-api:1.6.2'
+    compile "com.sun.mail:javax.mail:1.6.2"
```
Do these two lines of code provide different classes? Can we delete one of them, or they are both needed?

Valbod-I-am-a-Swiftie commented on Spring-server/src/main/java/hello/Application.java:
```diff
@@ -0,0 +1,30 @@
+                System.out.println(beanName);
+            }
+        };
+    }
```
Is this for debug? Can we delete it?

Valbod-I-am-a-Swiftie commented on Spring-server/src/main/java/hello/LoadDatabase.java:
```diff
@@ -0,0 +1,20 @@
+            repository.save(new User("sample", "pass",
+                    "sample@gmail.com", "qwerty",
+                    "smtp.google.com", 883,
+                    "imap.google.com", 666));
```
For the visibility i'd like to change "google.com" on "rambler.ru", for example, 'cause we don't have the Gmail-support for now :c

ValentinBodrov commented on README.md:
```diff
@@ -0,0 +1,17 @@
+#### Current features
+
+- None yet
+- But we have some uml-s and web-gui design ideas!
```
I think we have some features for now, so it could be complited

Egridnev replied:
yeah, we definitely need to update this file with our new features and set up description for uml-s in more appropriate way

Egridnev replied:
yeah, we definitely need to update this file with our new features and set up description for uml-s in more appropriate way

ValentinBodrov commented on Spring-server/build.gradle:
```diff
@@ -0,0 +1,52 @@
+dependencies {
+    compile "org.springframework.boot:spring-boot-starter-web"
+    compile 'org.springframework.boot:spring-boot-starter-undertow'
+    compile 'javax.mail:javax.mail-api:1.6.2'
+    compile "com.sun.mail:javax.mail:1.6.2"
```
Do these two lines of code provide different classes? Can we delete one of them, or they are both needed?

IReallyHateLogins replied:
Both needed, one is api, other is some core javax underneath, as i understand. Without first we won't have JavaMail classes, won't compile without second.

ValentinBodrov commented on Spring-server/src/main/java/hello/Application.java:
```diff
@@ -0,0 +1,30 @@
+                System.out.println(beanName);
+            }
+        };
+    }
```
Is this for debug? Can we delete it?

IReallyHateLogins replied:
Pretty much. We should correct this, agreed.

snowzhop replied:
Okay, we can just delete this

ValentinBodrov commented on Spring-server/src/main/java/hello/LoadDatabase.java:
```diff
@@ -0,0 +1,20 @@
+            repository.save(new User("sample", "pass",
+                    "sample@gmail.com", "qwerty",
+                    "smtp.google.com", 883,
+                    "imap.google.com", 666));
```
For the visibility i'd like to change "google.com" on "rambler.ru", for example, 'cause we don't have the Gmail-support for now :c

IReallyHateLogins replied:
Not strictly required, but could be done.

Egridnev replied:
why are we using magic numbers for port numbers? we should remowe them asap and use specific variables instead

Egridnev replied:
why are we using magic numbers for port numbers? we should remowe them asap and use specific variables instead``

IReallyHateLogins replied:
It's really a testing setup, we might as well remove this default user profile for now.

ValentinBodrov commented on Spring-server/src/test/java/hello/HelloControllerTest.java:
```diff
@@ -0,0 +1,39 @@
+//                 .andExpect(status().isOk())
+//                 .andExpect(content().string(equalTo("PRIVET PACANI!!!\n")));
+//     }
+// }
```
These pretty autotests should be completed lol

Egridnev replied:
why do we even add this if it is not ready yet?

IReallyHateLogins replied:
It was working for, like, 2 commits and then we really didn't want to figure out how to change it into both useful and working state.

ValentinBodrov commented on Spring-server/src/test/java/hello/HelloControllerTest.java:
```diff
@@ -0,0 +1,39 @@
+//     }
+
+//     @Test
+//     public void getHuet() throws Exception {
```
delete this pls

snowzhop replied:
No.

Egridnev replied:
Same as before

dahah-babah replied:
lol

ValentinBodrov reviewed (dismissed):
### Well done, team
We did it, the result is very nice. We should make some style-fixes (removing wildcards and some empty lines)

Please, check my comments in the changed files

IReallyHateLogins commented on Spring-server/src/main/java/hello/PostMessage.java:
```diff
@@ -0,0 +1,158 @@
+package hello;
+
+import javax.mail.*;
```
Missed this one in style fix

IReallyHateLogins commented on .gitignore:
```diff
@@ -0,0 +1,11 @@
+Spring-server/.idea/
+*.iml
+Spring-server/build/
+Spring-server/src/main/resources/public
```
Minor, but public resources ignore can be safely removed

Egridnev replied:
> Minor, but public resources ignore can be safely removed

Why?

IReallyHateLogins replied:
Because before I changed bulidsctipt, we copied assets into this path, now we copy them directly into the build folder.
Though there is another question, ideally we do not save resources and config in git, partially because we expose our ssl key, while not important and rather convenient now, we should be more careful if we get a normal signed certificate + key pair.

IReallyHateLogins commented on Spring-server/config/application.properties:
```diff
@@ -0,0 +1,13 @@
+server.port = 8443
+server.address = localhost
+
+# security.require-ssl = true 
```
This comment seems unnecessary

IReallyHateLogins commented on Spring-server/build.gradle:
```diff
@@ -0,0 +1,52 @@
+    compile "com.sun.mail:javax.mail:1.6.2"
+    implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
+    runtimeOnly 'com.h2database:h2'
+    // compile("org.springframework.boot:spring-boot-starter-security")
```
I think we can remove this commented import

IReallyHateLogins reviewed (commented):
So far:

 * One wildcard import

 * Fully commented out code in test files

 * Bad language in commented test function name

* Some seemingly redundant commented out imports and configs

 * Readme needs an update

 * Frontend needs to be reviewed by frontend developer.

 * Useless debug output of beans names

 * Maybe we should make default user entry a little more meaningful

snowzhop reviewed (dismissed):
I don't see any issues
I think we can merge it

Egridnev commented on assets/js/main.js:
```diff
@@ -0,0 +1 @@
+alert("vs code");
```
i dont think this line contains any useful code for project, so it should be removed

Egridnev commented on assets/js/auth.js:
```diff
@@ -0,0 +1,20 @@
+var username;
+var password;
+
+// var btnLogin = document.getElementById('btn');
```
we need to keep master branch clean, so i hardly recommend to remove all commented code before pull request, even if we plan to use it in future

Egridnev commented on assets/index.html:
```diff
@@ -2,9 +2,29 @@
+    </header>
+
+    <body>
+      hellow
```
Wait...Is this file basically empty and contains header and body with just one string?
Or i just don`t know anything about vue?

Egridnev commented on Spring-server/src/main/java/hello/PostMessage.java:
```diff
@@ -0,0 +1,158 @@
+package hello;
```
Why package name is still hello?
it`s a master pull request we need to keep it clean in all ways, especially for naming

IReallyHateLogins replied:
Ok, agreed

Egridnev reviewed (changes requested):
i dont think we should merge in this particular state, before we need to resolve the following issues:

- [x] File and package naming
- [x] Commented lines of code wich are not needed
- [x] Fully commented files wich are not needed
- [x] Some commit naming and description
- [x] **Magic numbers!!!!**

IReallyHateLogins committed:
Removed files used for testing

IReallyHateLogins committed:
Fixed some minor code issuses in backend

These changes should fix most of code quality and review issues in backend part of project

 - Corrected package name

 - Corrected rest controller name

 - Changed connectors class and variables names to be according to Google Java Style Guide - 5. Naming - 5.3Camel Case: defined

 - Removed some commented code and unneded imports

 - Removed debug output of beans names to console

 - Fixed wildcard imports

 - Changed variables and setters/getters plasement logic for some classes

 - Some other minor improvements

IReallyHateLogins committed:
App properties commented property removed

IReallyHateLogins committed:
Build script improvements

 - Replaced depricated methods for naming jar with currently correct ones

 - Moved compatibility and version higher, questionable

 - Corrected copy task to be more consistant and not skipping without reason

IReallyHateLogins committed:
Updated readme to reflect current project version

IReallyHateLogins committed:
Gitignore cleanup

dahahORbabah commented:
i was here

IReallyHateLogins committed:
Merge pull request #16 from Valbod-I-am-a-Swiftie/fix-sprint1-stable-issues

Fix sprint1 stable issues

Daniel1898 commented on Spring-server/src/main/java/ikpi63holding/postclient/ApiController.java:
```diff
@@ -0,0 +1,125 @@
+    }
+
+    @DeleteMapping("/mail/{id}")
+    public void getMail(@PathVariable Long id, @RequestBody PostMessage message) throws Exception {
```
I think we should rename this function "deleteMail"

ValentinBodrov replied:
I agree

IReallyHateLogins replied:
Yea, I was making this mapping for testing, so didn't care about correct naming, should fix

Daniel1898 commented on Spring-server/src/main/java/ikpi63holding/postclient/ImapConnector.java:
```diff
@@ -0,0 +1,140 @@
+    public List<String> getFolders() throws MessagingException {
+        return getAllFolders(this.store.getDefaultFolder());
+    }
+
```
What difference between getFolders and getAllFolders? I think getFolders is unnecessary.

IReallyHateLogins replied:
Get all folders is an internal method operaiting with JavaMail Folder object,
since I don't want to move any JavaMail object outside our mail and connector classes, I exposed this functionality through  parameterless method, inside witch I use store and get it's root folder to pass next.

Maybe we should think about correct naming, but I insist that both methods are serving a purpose.

Daniel1898 reviewed (commented):
Good job. But I have some comments about this.

ValentinBodrov reviewed (commented):
**ApiController.java**: we should rename the "getMail()" into "deleteMail()"
The other files look ready to merge

IReallyHateLogins committed:
Fixed controller delete mail method name

IReallyHateLogins committed:
Merge pull request #18 from Valbod-I-am-a-Swiftie/Sprint1-stable-patch-1

Fixed controller delete mail method name

ValentinBodrov reviewed (dismissed):
Now we are ready to merge it with master-branch, I think

snowzhop reviewed (dismissed):
Well, now we have a clean code. I'm approve it

dahah-babah committed:
Deleted main.js

IReallyHateLogins committed:
Merge pull request #19 from Valbod-I-am-a-Swiftie/remove-js-patch

Deleted main.js

dahah-babah committed:
Update auth.js

IReallyHateLogins committed:
Merge pull request #20 from Valbod-I-am-a-Swiftie/dahahORbabah-patch-1

Update auth.js

Egridnev reviewed (approved):
Well done, all issues are fixed

snowzhop reviewed (approved):
Well done C:

ValentinBodrov reviewed (approved):
###Finally, we did it!
I think, we can finally merge
